media/api_test.go

```diff
@@ -0,0 +1,249 @@
+// This file is part of the go-meta library.
+//
+// Copyright (C) 2017 JAAK MUSIC LTD
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// If you have any questions please contact yo@jaak.io
+
+package media
+
+import (
+	"testing"
+
+	"github.com/meta-network/go-meta/cwr"
+	"github.com/meta-network/go-meta/ern"
+	"github.com/meta-network/go-meta/identity"
+	"github.com/meta-network/go-meta/musicbrainz"
+	"github.com/meta-network/go-meta/testutil"
+	"github.com/meta-network/go-meta/testutil/index"
+)
+
+// TestResolver tests resolving META Media API queries.
+func TestResolver(t *testing.T) {
+	// create test indexes
+	store, cleanup := testutil.NewTestStore(t)
+	defer cleanup()
+	ernIndex, _ := testindex.GenerateERNIndex(t, "../ern", store)
+	defer ernIndex.Close()
+	cwrIndex, _ := testindex.GenerateCWRIndex(t, "../cwr", store)
+	defer cwrIndex.Close()
+	musicBrainzIndex, _, _ := testindex.GenerateMusicBrainzIndex(t, "../musicbrainz", store)
+	defer musicBrainzIndex.Close()
+
+	// create an ISWC -> ISRC mapping in the MusicBrainz index
+	link := &musicbrainz.RecordingWorkLink{
+		ISRC: "CASE00000001",
+		ISWC: "T1234567890",
+	}
+	linkObj, err := store.Put(link)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := musicBrainzIndex.Exec(
+		`INSERT INTO recording_work (object_id, isrc, iswc) VALUES ($1, $2, $3)`,
+		linkObj.Cid().String(), link.ISRC, link.ISWC,
+	); err != nil {
+		t.Fatal(err)
+	}
+
+	// create the resolver
+	resolver := &Resolver{
+		Ern:         ern.NewResolver(ernIndex.DB, store),
+		Cwr:         cwr.NewResolver(cwrIndex.DB, store),
+		MusicBrainz: musicbrainz.NewResolver(musicBrainzIndex.DB, store),
+		Store:       store,
+		IDStore:     identity.NewMemoryStore(),
+	}
```

Comment:
why is identity an `identity.MemoryStore`, thus different from `Resolver.Store` (which is disk)

Reply:
The identity store is distinct from object storage, identities will be stored in smart contracts whereas objects are stored in Swarm. There is an open task to implement a persistent identity store.